build/nested-cluster/BUILD

```diff
@@ -0,0 +1,49 @@
+package(default_visibility = ["//visibility:public"])
+
+load("@io_bazel_rules_docker//docker:docker.bzl", "docker_build", "docker_bundle")
+
+docker_build(
+    name = "nested-kubelet-build",
```

Comment:
nit: maybe follow the naming pattern of `foo-internal` and `foo` like is used in `build/BUILD`?

then you do `bazel build build/nested-cluster:nested-kubelet` or `bazel build build/nested-cluster:encapsulated-cluster` which is a little bit clearer IMO.

Reply:
we may also want to consider marking only the `docker_bundle`s with public visibility?